test/e2e/user_quota_tracing/user_quota_tracing_test.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+	siCommon "github.com/apache/yunikorn-scheduler-interface/lib/go/common"
+)
+
+const (
+	NANESPACE_LENGTH  = 10
+	WAIT_INTERVAL     = 60
+	DEFAULT_PARTITION = "default"
+)
+
+var _ = Describe("QuotaTracking: Two leaf queus for two groups", func() {
+	var ns string
+	BeforeEach(func() {
+		ns = "ns-" + common.RandSeq(NANESPACE_LENGTH)
+		By(fmt.Sprintf("Create namespace: %s", ns))
+		var ns1, err1 = kClient.CreateNamespace(ns, nil)
+		Ω(err1).NotTo(HaveOccurred())
+		Ω(ns1.Status.Phase).To(Equal(v1.NamespaceActive))
+	})
+
+	It("User qauta trace with 3 users and 2 groups", func() {
+		groups := []string{"group1", "group2"}
+		users := []string{"user1", "user2", "user3"}
+		queuePath := []string{"root", "root.group1_resources", "root.group2_resources"}
+		zeroPod := map[string]int64{siCommon.CPU: 0, siCommon.Memory: 0}
+		onePod := map[string]int64{siCommon.CPU: 100, siCommon.Memory: 50}
+		twoPods := map[string]int64{siCommon.CPU: 200, siCommon.Memory: 100}
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+			queuesConfigs := []struct {
+				partition, parentQueue, QueueName string
+			}{
+				{DEFAULT_PARTITION, queuePath[0], "group1_resources"},
+				{DEFAULT_PARTITION, queuePath[0], "group2_resources"},
+			}
+			for _, queueConfig := range queuesConfigs {
+				By(fmt.Sprintf("Add child queue %s to the parent queue %s", queueConfig.QueueName, queueConfig.parentQueue))
+				config := configs.QueueConfig{
+					Name: queueConfig.QueueName,
+				}
+				if err := common.AddQueue(sc, queueConfig.partition, queueConfig.parentQueue, config); err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+
+		/*
+		*  groups: group1, group2
+		*  users: user1, user2, user3
+		*  user1 -> group1
+		*  user2 -> group2
+		*  user3 -> group1, group2
+		 */
+		configs := []struct {
+			AppID, Queue, UserName string
+			GroupsNames            []string
+		}{
+			{"teacher-app-01", queuePath[1], users[0], []string{groups[0]}},
+			{"students-app-01", queuePath[2], users[1], []string{groups[1]}},
+			{"assistant-app-01", queuePath[2], users[2], []string{groups[1]}},
+			{"assistant-app-02", queuePath[1], users[2], []string{groups[0]}},
+		}
+		for _, config := range configs {
+			By(fmt.Sprintf("Deploy the sleep app %s to the %s namespace", config.AppID, ns))
+			sleepObj, podErr := k8s.InitSleepPod(k8s.SleepPodConfig{
+				AppID:  config.AppID,
+				NS:     ns,
+				Labels: map[string]string{"queue": config.Queue},
+			})
+			sleepObj.ObjectMeta.Annotations["yunikorn.apache.org/user.info"] = fmt.Sprintf("{username:%s, groups:{%s}}", config.UserName, config.GroupsNames)
```

Comment:
This will be rejected by the admission controller unless the testcase (or rather the user which the TC uses, which is "kubernetes-admin" in case of KIND) is allowed to do it. A new entry is necessary in the configmap:
`admissionController.accessControl.externalUsers: "kubernetes-admin"`

You can look at the existing admission tests to see how it's configured.

Reply:
Thanks for comments.
I will check the the way to configure external users in admin e2e test